apps/core/src/components/affine/auth/sign-in.tsx

```diff
@@ -37,17 +37,20 @@ export const SignIn: FC<AuthPanelProps> = ({
     }
 
     setIsValidEmail(true);
-    const res = await verifyUser({ email: email });
+    const { user } = await verifyUser({ email });
 
     setAuthEmail(email);
-    if (res?.user) {
-      signIn('email', {
-        email: email,
-        callbackUrl: buildCallbackUrl('signIn'),
-        redirect: true,
-      }).catch(console.error);
-
-      setAuthState('afterSignInSendEmail');
+    if (user) {
+      if (user.hasPassword) {
+        setAuthState('signInWithPassword');
+      } else {
+        signIn('email', {
+          email: email,
+          callbackUrl: buildCallbackUrl('signIn'),
+          redirect: true,
+        }).catch(console.error);
+        setAuthState('afterSignInSendEmail');
```

Comment:
should this one be put into `signIn.then`?

Reply:
I'm not sure, it's the original implementation

Reply:
/cc @QiShaoXuan